iris-location-service/src/main/java/iris/location_service/jsonrpc/LocationRPCImpl.java

```diff
@@ -29,90 +35,163 @@
 	private final @NotNull HealthEndpoint healthEndpoint;
 
 	@Override
 	public String postLocationsToSearchIndex(JsonRpcClientDto client, List<LocationInformation> locationList) {
 
-		var listOfInvalidAddresses = locationService.addLocations(client.getName(), locationList);
+		if (!isJsonRpcClientDtoInputValid(client)) {
+			throw new ResponseStatusException(HttpStatus.BAD_REQUEST, ErrorMessageHelper.INVALID_INPUT_EXCEPTION_MESSAGE);
+		}
+
+		List<LocationInformation> locationListValidated = locationInformationInputValidated(locationList);
+
+		var listOfInvalidAddresses = locationService.addLocations(client.getName(), locationListValidated);
 
 		var ret = "OK";
 		var logRes = ret;
 
 		if (!listOfInvalidAddresses.isEmpty()) {
 
-			ret = listOfInvalidAddresses.stream()
-					.collect(Collectors.joining(", ", "Invalid Locations detected: ", ""));
+			ret = listOfInvalidAddresses.stream().collect(Collectors.joining(", ", "Invalid Locations detected: ", ""));
 			logRes = "Invalid Locations detected";
 		}
 
-		log.debug("JSON-RPC - Post locations for client: {} (locations: {}) => result: {}", client.getName(),
-				locationList.size(), logRes);
+		log.debug("JSON-RPC - Post locations for client: {} (locations: {}) => result: {}", client.getName(), locationList.size(), logRes);
 
 		return ret;
 	}
 
 	@Override
 	public List<LocationOverviewDto> getProviderLocations(JsonRpcClientDto client) {
 
+		validateInputForGetProviderLocations(client);
+
 		var providerLocations = locationService.getProviderLocations(client.getName());
 
-		log.debug("JSON-RPC - Get provider locations for client: {} => returns {} locations", client.getName(),
-				providerLocations.size());
+		log.debug("JSON-RPC - Get provider locations for client: {} => returns {} locations", client.getName(), providerLocations.size());
 
 		return providerLocations;
 	}
 
 	@Override
 	public String deleteLocationFromSearchIndex(JsonRpcClientDto client, String locationId) {
+		validateInputForDeleteLocationFromSearchIndex(client, locationId);
 
 		var ret = "NOT FOUND";
 
 		if (locationService.deleteLocation(client.getName(), locationId))
 			ret = "OK";
 
-		log.debug("JSON-RPC - Delete location for client: {}; locationId: {} => result: {}", client.getName(),
-				obfuscate(locationId), ret);
+		log.debug("JSON-RPC - Delete location for client: {}; locationId: {} => result: {}", client.getName(), obfuscate(locationId), ret);
 
 		return ret;
 	}
 
 	@Override
 	public LocationQueryResult searchForLocation(JsonRpcClientDto client, String searchKeyword, PageableDto dto) {
 
-		var pageable = PageRequest.of(dto.getPage(), dto.getSize(),
-				dto.getSortBy() != null
-						? Sort.by(dto.getDirection(), dto.getSortBy())
-						: Sort.unsorted());
+		if (ValidationHelper.isPossibleAttack(searchKeyword, EXCEPTION_MESSAGE_SEARCH_KEYWORD + searchKeyword)) {
+			throw new ResponseStatusException(HttpStatus.BAD_REQUEST, ErrorMessageHelper.INVALID_INPUT_EXCEPTION_MESSAGE);
```

Comment:
Why do you pass the finished message to the method? This could be assembled according to a uniform pattern in the method, which receives the value and the field name for this check and logging.

Reply:
That was first idea as well, but if we do so we have the problem that we suddenly could display a password which we probably do not want to do even in case of a possible error.

Reply:
Ok, that's an argument, but do we have security or privacy related data in the LocationService?

Reply:
good point let me check and adjust if neccessary

Reply:
This is probably a matter of opinion, but I would, if relevant, provide another helper method `*LogObfuscated` that replaces all characters with "*" except the suspicious ones for relevant cases. Then you don`t have to worry about assembling the message several times and esp. about obfuscating.